test/Swan.Test/TableContextGenericTest.cs

```diff
@@ -0,0 +1,572 @@
+            EndDate = DateTime.Now,
+            IsActive = true,
+            Name = "Project ONE",
+            ProjectScope = "My Scope",
+            ProjectType = ProjectTypes.Exciting,
+            StartDate = DateTime.Now.AddMonths(-1)
+        });
+
+        Project project = new Project()
+        {
+            ProjectId = 1,
+            CompanyId = 1,
+            EndDate = DateTime.Now,
+            IsActive = true,
+            Name = "Project Updated",
+            ProjectScope = "My Scope",
+            ProjectType = ProjectTypes.Exciting,
+            StartDate = DateTime.Now.AddMonths(-1)
+        };
+
+        var projectUpdated = await table.UpdateOneAsync(project);
+
+        Assert.AreEqual(1, projectUpdated);
+    }
+
+    [Test]
+    public void UpdateManyRetrunsNumberOfRowsAffected()
+    {
+        var conn = new SqliteConnection("Data Source=:memory:");
+        conn.EnsureConnected().TableBuilder<Project>("Projects").ExecuteDdlCommand();
+
+        var table = conn.Table<Project>("Projects");
+
+        List<Project> projects = new List<Project>
+        {
+            new Project()
+            {
+                CompanyId = 1,
+                EndDate = DateTime.Now,
+                IsActive = true,
+                Name = "Project ONE",
+                ProjectScope = "My Scope",
+                ProjectType = ProjectTypes.Exciting,
+                StartDate = DateTime.Now.AddMonths(-1)
+            },
+            new Project()
+            {
+                CompanyId = 2,
+                EndDate = DateTime.Now,
+                IsActive = true,
+                Name = "Project TWO",
+                ProjectScope = "My Scope",
+                ProjectType = ProjectTypes.Boring,
+                StartDate = DateTime.Now.AddMonths(-1)
+            }
+        };
+
+        List<Project> projectsUpdated = new List<Project>
+        {
+            new Project()
+            {
+                ProjectId = 1,
+                CompanyId = 1,
+                EndDate = DateTime.Now,
+                IsActive = true,
+                Name = "Project ONE Updated",
+                ProjectScope = "My Scope",
+                ProjectType = ProjectTypes.Exciting,
+                StartDate = DateTime.Now.AddMonths(-1)
+            },
+            new Project()
+            { 
+                ProjectId = 2,
+                CompanyId = 2,
+                EndDate = DateTime.Now,
+                IsActive = true,
+                Name = "Project TWO Updated",
+                ProjectScope = "My Scope",
+                ProjectType = ProjectTypes.Boring,
+                StartDate = DateTime.Now.AddMonths(-1)
+            }
+        };
+
+        table.InsertMany(projects);
+        var updatedCount = table.UpdateMany(projectsUpdated);
+        var updatedTable = table.Query().ToList();
+
+        Assert.AreEqual(projectsUpdated.Count, updatedCount);
+        Assert.AreEqual("Project ONE Updated", updatedTable[0].Name);
```

Comment:
You can also reference your original list instead of the string in this case, something like
`Assert.AreEqual(projects[0].Name, updatedTable[0].Name)`
same for other tests